x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/AttributeSet.java

```diff
@@ -113,7 +113,7 @@ public <T> T[] toArray(T[] a) {
 
     @Override
     public boolean add(Attribute e) {
-        return delegate.put(e, PRESENT) != null;
+        return delegate.put(e, PRESENT) == null;
     }
```

Comment:
Changed this to follow the Set conventions. It was probably a typo.

We should change it in the QL counterpart too. I haven't done it yet, as to avoid mixing more things here

Reply:
I might pull this one character change out into it's own PR.....

Reply:
Let me be clear - I'd probably pull this into it's own PR because there are like 1232134 callers of this method. Do they use the return value? That's hard to tell because this is a collection implementation. If you land it in one PR, even as a one line change. Or a one line change with a single test, then we will have an easier time tracking tricky changes to it.

I'm just imagining `git bisect` pointing to a PR that does *just* this thing vs a PR that does, well, all of CATEGORIZE.

Reply:
https://github.com/elastic/elasticsearch/pull/117367

Reply:
PS: Absolutely, I'm with you here, it makes all the sense!